libmachine/provision/debian.go

```diff
@@ -79,6 +77,28 @@ func (provisioner *DebianProvisioner) Package(name string, action pkgaction.Pack
 		}
 	}
 
+	// handle the new docker-engine package; we can probably remove this
+	// after we have a few versions
+	if action == pkgaction.Upgrade && name == "docker-engine" {
+		// run the force remove on the existing lxc-docker package
+		// and remove the existing apt source list
+		// also re-run the get.docker.com script to properly setup
+		// the system again
+
+		commands := []string{
+			"rm /etc/apt/sources.list.d/docker.list || true",
+			"apt-get remove -y lxc-docker || true",
+			"curl -sSL https://get.docker.com | sh",
+		}
+
+		for _, cmd := range commands {
+			command := fmt.Sprintf("DEBIAN_FRONTEND=noninteractive sudo -E %s", cmd)
```

Comment:
What's the reason for the `sudo -E` here?

If it's just to make sure `DEBIAN_FRONTEND` is passed through, you should be able to swap them and accomplish the same, ie: `sudo DEBIAN_FRONTEND=noninteractive %s`.